modules/java/native/java_machine.c

```diff
@@ -113,6 +113,12 @@ void
 java_machine_unref(JavaVMSingleton *self)
 {
   g_assert(self == global_jvm);
+  /* The last reference is always hold by a AH_SHUTDOWN app hook, when there is no java configured.
+   * The only way to get ride of it to shutdown syslog-ng.  */
+  if (g_atomic_counter_racy_get(&self->ref_cnt) == 1)
+    {
+      msg_warning("There is no need for the JVM, but it is not stoped, if you want to stop it, you have to restart syslog-ng");
```

Comment:
I'd rephrase this sentence to make it more understandable, for example:
The JVM instance has become unused, but it has not been stopped. Restart syslog-ng, if you want to stop it.

A side note: this warning message will be displayed every time you stop syslog-ng (not just on reloads), if I understand it correctly.

Reply:
(There is a JNI method called `DestroyJavaVM`, but it seems it is not guaranteed that it will stop the VM.)